src/components/LandingPageComponents/Stats.js

```diff
@@ -1,490 +1,465 @@
-import React, {useState, useEffect} from 'react'
+import React, { useState, useEffect } from 'react'
 import styled from 'styled-components'
+import { ApolloClient, InMemoryCache, gql } from '@apollo/client'
+import { ChainId } from '@swapr/sdk'
+
 import { gradients, breakpoints } from '../../utils/theme'
 import { StatsContent } from '../../utils/ui-constants'
 import { toClassName } from './../../utils/helper-functions'
 import Layout from './layout/Layout'
-import TextyAnim from 'rc-texty';
-
-import { ApolloClient, InMemoryCache, gql } from '@apollo/client';
+import TextyAnim from 'rc-texty'
+import { subgraphClients } from '../../apollo/client'
 
 const apiUrlList = [
-    'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-arbitrum-one-v3',
-    'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-xdai-v2',
-    'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-mainnet-v2'
+  subgraphClients[ChainId.ARBITRUM_ONE].link.options.uri,
+  subgraphClients[ChainId.XDAI].link.options.uri,
+  subgraphClients[ChainId.MAINNET].link.options.uri
 ]
 
 const tokensQuery = `{
     swaprFactories (first:1000) {
         txCount
         totalVolumeUSD
     }
-}`;
-
+}`
 
-const retrieveData = async (url) => {
+const retrieveData = async url => {
+  const client = new ApolloClient({
+    uri: url,
+    cache: new InMemoryCache()
+  })
```

Comment:
This client is created every time `retrieveData` is invoked. Hence, there's no caching really. You can move it up.

Reply:
Yup. This makes a lot of sense. Thank you Adam. I'll fix it.